pkg/controllers/monitoring/thanos-querier/controller.go

```diff
@@ -41,6 +42,30 @@ type resourceManager struct {
 	client.Client
 	scheme *runtime.Scheme
 	logger logr.Logger
+	thanos ThanosConfiguration
+}
+
+type ThanosConfiguration struct {
+	Image   string
+	Version string
+}
+
+func (t ThanosConfiguration) GetImageRef() string {
+	image := "quay.io/thanos/thanos"
+	if t.Image != "" {
+		image = t.Image
+	}
+	if t.Version == "" {
```

Comment:
I'd expect the --image arg to be a fully specified image name?

Reply:
Right now the operator accepts `--images` and `--versions`. The motivation is when a user wants to rely on the default location but with a specific version.
Though I guess in relation to OLM's `relatedImages` this is not a use case. I'll change this accordingly

Reply:
:+1: I was actually taking inspiration from p-o initially :laughing: